### PR TITLE
`tab` URL parameter in Wallet page

### PR DESCRIPTION
**Pull request by think-in-universe**
### Description

Resolve: https://github.com/near/near-wallet/issues/1877

1. if open https://near-wallet-pr-1895.onrender.com/?tab=nft, the `Collectibles` tab will be open
2. if `tab` is not set, or set to other values, the `Balance` tab will be open
3. when click on the tab `Collectibles`, URL will be changed to https://near-wallet-pr-1895.onrender.com/?tab=nft
4. when click on the tab `Balance`, URL will be changed to https://near-wallet-pr-1895.onrender.com

### Discussion

1. Is it OK to use the `tab` parameter? should we use other params like `token`?
2. Is it OK to use the `nft` value for `tab`? should we use `collectibles`?

### CODE REVIEW

**Comment by Patrick1904** [comment]
Thanks for adding this @think-in-universe - it's a nice detail worth supporting IMO. 

I think it'd be good to align the language like you suggested and call it whatever is shown on the UI. So in this case, only use `balances` and `collectibles`, for both state and URL param. 

We could also re-name the state to `const [view, setView] = useState(getView());`

Also nit but it'd be nice to declare `const location = props.history.location;` and move the `getTokenView (getView)` function to right above the `switchTokenView (switchView)` function just to make it feel a bit more organized.

**Comment by MaximusHaximus** [comment]
@think-in-universe Thanks for taking the initiative to work on this concept :) I think it'll be a nice addition!

I think that conceptually, this is part of our `Routing` stack, not an internal behaviour of the `Wallet` component.  I think we should avoid adding internal state management to the `Wallet` component, and instead handle this navigation at our Router level; this is how we handle everything that is driven via URL paths/parameters/queries that control major portions of the UI, and it keeps tight coupling with the `location` out of our presentational components (which is a good thing for testability and for keeping our display components simple).

We could change from using the component argument to configure the Route to using the render API instead.
https://reactrouter.com/web/api/Route/render-func

Since the `render()` function receives the Route properties (including location -- ref: https://reactrouter.com/web/api/Route/route-props), we can render the child component with different props, based on anything in the location object (such as parameters, paths, or query strings).

Then, by leveraging the router for this logic, changing the active tab will become a completely standard route/navigation call instead of us mucking around with local state in the `Wallet` component -- and since how the Wallet component functions would be props driven rather then internally inferred by inspecting router properties itself and then setting and checking locally managed state, we can later add tests to it that won't rely on the location being mocked when we render the component.

**Comment by think-in-universe** [comment]
@Patrick1904 @MaximusHaximus thanks for the suggstions :) 

updated, and please have a look.

File: src/components/wallet/Wallet.js [hunk 1]
```diff
@@ -306,26 +306,26 @@ export function Wallet() {
...
-                            className={classNames(['tab-collectibles', tokenView !== 'nonFungibleTokens' ? 'inactive' : ''])}
-                            onClick={() => setTokenView('nonFungibleTokens')}
+                            className={classNames(['tab-collectibles', tab !== 'collectibles' ? 'inactive' : ''])}
+                            onClick={() => setTab('collectibles')}
                         >
                             <Translate id='wallet.collectibles' />
                         </div>
                     </div>
-                    {tokenView === 'fungibleTokens' &&
+                    {!tab &&
```
Comment (Patrick1904) [inline src/components/wallet/Wallet.js:321]: To guard against an invalid tab param, e.g. `tab=blabla` that would result in neither balances nor collectibles being rendered, we can do:

```
{tab === 'collectibles' ?
    <NFTs tokens={sortedNFTs} />
    :
    <FungibleTokens
        balance={balance}
        tokensLoader={tokensLoader}
        sortedTokens={sortedTokens}
    />
}
```

And then just change `L309` to:
`className={classNames(['tab-balances', tab === 'collectibles' ? 'inactive' : ''])}`

Comment (think-in-universe) [reply src/components/wallet/Wallet.js]: Thanks Patrick! good catch. fixed now.

**Comment by think-in-universe** [comment]
@MaximusHaximus @Patrick1904 

please review whether my merge conflicts resolution is correct or not in https://github.com/near/near-wallet/pull/1895/commits/ee845ed3daa025fc4f4ccb374c9fc6ed0ce0fdaf

**Review by Patrick1904 — APPROVED** [review]

**Review by MaximusHaximus — APPROVED** [review]
@think-in-universe Thanks for the submission + updates to move things into the Router layer!

I noticed two issues:
1. We were completely creating the URL to navigate to when the `tab` was clicked, which had a side-effect of removing any `hash` value in the URL.
2. If someone called `setTab()` with an empty string, then we were deleting the _entire_ `query` portion of the URL _and_ the hash; switching tabs shouldn't necessarily remove other parts of the URL such as `hash` and other `query` parameters.

In the interest of time, given that we want to ship this today, I have pushed a commit to resolve both of these cases :)

Thanks again for your help!

**Comment by think-in-universe** [comment]
@MaximusHaximus thanks Daryl. nice fix, and great team work. 

We delivered a bunch of critical updates in a short period of time. That's really amazing and helpful. Thanks a lot everyone.

**Pull request merged**